micronaut-maven-plugin/src/main/java/io/micronaut/maven/openapi/OpenApiGenericMojo.java

```diff
@@ -0,0 +1,111 @@
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Locale;
+import java.util.Map;
+
+@Mojo(name = OpenApiGenericMojo.MOJO_NAME, defaultPhase = LifecyclePhase.GENERATE_SOURCES)
+public class OpenApiGenericMojo extends AbstractOpenApiMojo {
+    public static final String MOJO_NAME = "generate-openapi-sources";
+    public static final String CONFIGURATION_PROPERTIES = MICRONAUT_OPENAPI_PREFIX + ".generator.properties";
+
+    @Parameter(property = MICRONAUT_OPENAPI_PREFIX + ".generator.builder.classname")
+    protected String generatorClassName;
+
+    @Parameter(property = CONFIGURATION_PROPERTIES)
+    protected Map<String, String> properties;
+
+    @Override
+    protected boolean isEnabled() {
+        return generatorClassName != null;
+    }
+
+    @Override
+    protected void configureBuilder(MicronautCodeGeneratorBuilder builder) {
+        MicronautCodeGenerator<? extends GeneratorOptionsBuilder> generator = instantiateGenerator();
+        builder.forCodeGenerator(generator, config -> {
+            for (Map.Entry<String, String> entry : properties.entrySet()) {
+                String name = entry.getKey().substring(CONFIGURATION_PROPERTIES.length() + 1);
+                String value = entry.getValue();
+                invokeMethod(name, config, value);
+            }
+        });
+    }
+
+    private MicronautCodeGenerator<? extends GeneratorOptionsBuilder> instantiateGenerator() {
+        MicronautCodeGenerator<? extends GeneratorOptionsBuilder> generator;
+        try {
+            generator = (MicronautCodeGenerator<? extends GeneratorOptionsBuilder>) this.getClass()
+                    .getClassLoader()
+                    .loadClass(generatorClassName)
+                    .getDeclaredConstructor()
+                    .newInstance();
+        } catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException | ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+        return generator;
+    }
+
+    private static void invokeMethod(String name, GeneratorOptionsBuilder builder, String value) {
+        try {
+            String witherName = "with" + StringUtils.capitalize(name);
+            String setterName = "set" + StringUtils.capitalize(name);
+            Class<? extends GeneratorOptionsBuilder> builderClazz = builder.getClass();
+            var methods = builderClazz.getDeclaredMethods();
+            for (Method method : methods) {
+                if (invokeIfMatches(name, builder, value, witherName, setterName, method)) {
+                    return;
+                }
+            }
+            throw new RuntimeException("Unable to find a method on builder " + builderClazz + " with name '" + name + "' which accepts argument '" + value + "'");
```

Comment:
`RuntimeException` should not be propagated all the way through. You should throw either `MojoExecutionException` (for a `BUILD ERROR`, which means an actual exceptional situation)  or `MojoFailureException` (for a `BUILD FAILURE`, when the build is ok but failed due to tests, checks, prerequisites, etc).

If you still want to throw generic exceptions down here it's fine, but be sure to wrap them in `AbstractOpenApiMojo#execute`